pkg/restore/restore_new_test.go

```diff
@@ -721,6 +724,237 @@ func TestInvalidTarballContents(t *testing.T) {
 	}
 }
 
+// TestRestoreItems runs restores of specific items and validates that they are created
+// with the expected metadata/spec/status in the API.
+func TestRestoreItems(t *testing.T) {
+	tests := []struct {
+		name         string
+		restore      *velerov1api.Restore
+		backup       *velerov1api.Backup
+		apiResources []*test.APIResource
+		tarball      io.Reader
+		want         []*test.APIResource
+	}{
+		{
+			name:    "metadata other than namespace/name/labels/annotations gets removed",
+			restore: defaultRestore().Restore(),
+			backup:  defaultBackup().Backup(),
+			tarball: newTarWriter(t).
+				addItems("pods",
+					test.NewPod("ns-1", "pod-1",
+						test.WithLabels("key-1", "val-1"),
+						test.WithAnnotations("key-1", "val-1"),
+						test.WithClusterName("cluster-1"),
+						test.WithFinalizers("finalizer-1")),
+				).
+				done(),
+			apiResources: []*test.APIResource{
+				test.Pods(),
+			},
+			want: []*test.APIResource{
+				test.Pods(
+					test.NewPod("ns-1", "pod-1",
+						test.WithLabels("key-1", "val-1", "velero.io/backup-name", "backup-1", "velero.io/restore-name", "restore-1"),
+						test.WithAnnotations("key-1", "val-1"),
+					),
+				),
+			},
+		},
+		{
+			name:    "status gets removed",
+			restore: defaultRestore().Restore(),
+			backup:  defaultBackup().Backup(),
+			tarball: newTarWriter(t).
+				addItems("pods",
+					&corev1api.Pod{
+						TypeMeta: metav1.TypeMeta{
+							APIVersion: "v1",
+							Kind:       "Pod",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Namespace: "ns-1",
+							Name:      "pod-1",
+						},
+						Status: corev1api.PodStatus{
+							Message: "a non-empty status",
+						},
+					},
+				).
+				done(),
+			apiResources: []*test.APIResource{
+				test.Pods(),
+			},
+			want: []*test.APIResource{
+				test.Pods(
+					test.NewPod("ns-1", "pod-1", test.WithLabels("velero.io/backup-name", "backup-1", "velero.io/restore-name", "restore-1")),
+				),
+			},
+		},
+		{
+			name:    "object gets labeled with full backup and restore names when they're both longer than 63 characters",
```

Comment:
Isn't this the shorter case, not the longer one?

Reply:
yep, will update